# HG changeset patch
# User John D <john.donoghue@ieee.org>
# Date 1505603593 14400
#      Sat Sep 16 19:13:13 2017 -0400
# Branch stable
# Node ID b02d953d4b9ae3e953e7d247dc9b6a03740c4840
# Parent  16fae04366b2eaeb6422967fd084213cbd567e6f
qterminal(win32): mark null characters as spaces when copying to clipboard

* libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp:
  (QConsolePrivate::getSelection): if character is null, set it to a space character

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -538,16 +538,18 @@ QString QConsolePrivate::getSelection (v
                 {
                   if (lastNonSpace >= 0)
                     selection.truncate (lastNonSpace);
                   selection.append ('\n');
                   lastNonSpace = selection.length ();
                 }
 
               QChar c (buf[i].Char.UnicodeChar);
+              if (c.isNull())
+                c = QChar(QChar::Space);
 
               selection.append (c);
               if (! c.isSpace ())
                 lastNonSpace = selection.length ();
             }
 
           if (lastNonSpace >= 0)
             selection.truncate (lastNonSpace);
